src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs

```diff
@@ -137,6 +156,27 @@ private void InitializeFromArray(JsonNode?[] items)
             _list = list;
         }
 
+        private void InitializeFromSpan(ReadOnlySpan<JsonNode?> items)
+        {
+            List<JsonNode?> list = new(items.Length);
+
+#if NET8_0_OR_GREATER
+            list.AddRange(items);
+#else
+            foreach (JsonNode? item in items)
+            {
+                list.Add(item);
+            }
+#endif
+
+            for (int i = 0; i < items.Length; i++)
+            {
+                items[i]?.AssignParent(this);
+            }
```

Comment:
For a bit of defense in depth, probably better to do this as:
```suggestion
            for (int i = 0; i < list.Count; i++)
            {
                list[i]?.AssignParent(this);
            }
```
on the off chance that the elements of the source were being mutated concurrently with this call (which wouldn't be supported, but we could end up with difficult to diagnose behaviors)